src/tqec/_cli/subcommands/run_example.py

```diff
@@ -0,0 +1,178 @@
+            ),
+            nargs="*",
+            type=int,
+        )
+        parser.add_argument(
+            "--add-detectors",
+            help="Whether to add detectors to the circuits.",
+            action="store_true",
+        )
+        parser.add_argument(
+            "--generate-only",
+            help="Whether to run generation only.",
+            action="store_true",
+        )
+        parser.add_argument(
+            "--code-style",
+            help="Use the CSS or ZXXZ code style.",
+            choices=["css", "zxxz"],
+        )
+        parser.add_argument(
+            "--include-z-basis",
+            help="Whether to include the Z basis in the simulation.",
+            action="store_true",
+        )
+        parser.set_defaults(func=RunExamplesTQECSubCommand.execute)
+
+    @staticmethod
+    @override
+    def execute(args: argparse.Namespace) -> None:
+        # Parse args
+        logging.info("Parsing arguments.")
+        out_dir: Path = args.out_dir.resolve()
+        obs_indices: list[int] = args.obs_include
+        style: str = args.code_style
+        z_basis: bool = args.include_z_basis
+        ks: list[int] = args.k
+        ps: list[float] = args.p
+
+        # Set up the output directories
+        logging.info("Setting up output directories.")
+        if not out_dir.exists():
+            out_dir.mkdir(parents=True)
+
+        simulations_out_dir = out_dir / "simulations"
+        simulations_out_dir.mkdir(exist_ok=True)
+        plots_out_dir = out_dir / "plots"
+        plots_out_dir.mkdir(exist_ok=True)
+
+        logging.info("Generating CNot block graph and zx graph.")
+        block_graph = logical_cnot_block_graph(z_basis)
+        zx_graph = block_graph.to_zx_graph()
+
+        # observables to a subdirectory
+        logging.info("Find the observables.")
+        observables, correlation_surfaces = block_graph.get_abstract_observables()
+        if not obs_indices:
+            obs_indices = list(range(len(observables)))
+        if max(obs_indices) >= len(observables):
+            raise ValueError(
+                f"Found {len(observables)} observables,"
+                + f"but requested indices up to {max(obs_indices)}."
+            )
+
+        block_builder = CSS_BLOCK_BUILDER if style == "css" else ZXXZ_BLOCK_BUILDER
+        substitution_builder = (
+            CSS_SUBSTITUTION_BUILDER if style == "css" else ZXXZ_SUBSTITUTION_BUILDER
+        )
+
+        logging.info("Start the simulation, this might take some time.")
+        stats = start_simulation_using_sinter(
+            block_graph,
+            ks,
+            ps,
+            NoiseModel.uniform_depolarizing,
+            block_builder=block_builder,
+            substitution_builder=substitution_builder,
+            observables=[observables[i] for i in obs_indices],
+            num_workers=20,
+            max_shots=10_000_000,
+            max_errors=5_000,
```

Comment:
Maybe add parameters in the CLI for these, or use a better default value for `num_workers` (e.g., `multiprocessing.cpu_count()`?)

Reply:
I wasn't sure how much to add for the cli. If this was a general `run-simulation` command I'd agree but since this is just generating the example I don't think this is of much use in the cli. I'll change the default though.

Reply:
I think that at least the number of CPU should be changed to something that depends on the host machine. Running 20 processes on a dual-core CPU makes no sense.